add reference blocksize and checkpointing to VCF (#435)
samtools/hts-specs: VCFv4.3.tex

##pedigreeDB=URL
\end{verbatim}

> \subsubsection{Reference block checkpoint}

Reviewer: This should be placed **after** the final sentence (“See [PedigreeInDetail] for details”) of the preceding section.

Given the ability to interpret missing genotypes as either truly missing or as part of a reference block (see RBS in Genotype tags below), it can be useful to limit the genomic distance required to scan in order to find the "top" of the reference block. 
To enable this, one may specify a Reference Block Checkpoint scheme:
\begin{verbatim}

> ##REFERENCE_BLOCK=<CHECKPOINT=1000>

Reviewer: This is invalid as a structured meta-information line, because it does not have an `ID=…` subfield.

Are there plans to expand this in future with additional subfields within the `<…>`? If not, I'd suggest reworking this as a plain unstructured meta-information line, e.g.,

```
##REFERENCE_BLOCK_CHECKPOINT=1000
```

> ##REFERENCE_BLOCK=<CHECKPOINT=1000>

Reviewer 2: ```suggestion
##REFERENCE_BLOCK_CHECKPOINT=1000
```

just reifying this as a one-click GitHub suggestion.

      PL		& G		& Integer	& Phred-scaled genotype likelihoods rounded to the closest integer \\
      PQ		& 1		& Integer	& Phasing quality \\
      PS		& 1		& Integer	& Phase set \\

>       RBS		& 1		& Integer	& Reference Block Size\\ 

Reviewer 3: ```suggestion
      RBS		& 1		& Integer	& Reference block size\\ 
```

  the confidence other attributes of the missing genotypes are the same as that in the anchor genotype (the one with the RBS value). 
  Clearly, this can only be used when the genotype in the anchor variant is reference.
  The numerical value of RBS is the difference between the last position (inclusive) of the reference block and POS. 
  Missing genotypes (`.') that are not covered by a reference block are to be interpreted as missing, i.e. no information is known about the site. 
  To disambiguate a `.'  between being truly missing and part of a reference block, one would therefore need to "look up" and find the previous RBS FORMAT value in that sample.
  In addition, any non-missing value (including `.:.' or `./.') would effectively break a reference block, and should be treated as a violation of the specification if RBS is specified, or an implicit end of the block if RBS is unknown.
  When reading the file from top to bottom, an implementation can simply remember what the RBS is for each sample, however when using the index to ``seek" to a particular point of the reference, one may need to seek to an unknown location in the file. 

>   To assist in seeking, the \verb!##REFERENCE_BLOCK! header line may define the \verb!CHECKPOINT! multiple at which a reference block will be included for all samples. In the presence of a checkpoint value, an implementation can read back from the last checkpoint and on and be assured that it will find a reference block that overlaps the current position, if it exists. 

Reviewer 2: ```suggestion
  To assist in seeking, the \verb!##REFERENCE_BLOCK_CHECKPOINT! header line defines a multiple at which a reference block will be included for all samples. In the presence of a checkpoint value, an implementation can read back from the last checkpoint and on and be assured that it will find a reference block that overlaps the current position, if it exists. 
```
